configuration/nfs.md

```diff
@@ -0,0 +1,304 @@
+```
+
+The filename `yp` refers to Yellow Pages, the former name of NIS.
+
+### Portmap lockdown (optional)
+
+Add the following line to `/etc/hosts.deny`:
+
+```
+rpcbind mountd nfsd statd lockd rquotad : ALL
+```
+
+By blocking all clients first, only clients in `/etc/hosts.allow` (added below) will be allowed to access the server.
+
+Consider adding the following line to `/etc/hosts.allow`:
+
+```
+rpcbind mountd nfsd statd lockd rquotad : <list of IPs>
+```
+
+where `<list of IPs>` is a list of the IP addresses of the server and all clients. These have to be IP addresses because of a limitation in `rpcbind`. Note that if you have NIS set up, you can just add these to the same line.
+
+### Package installation and configuration
+
+Install the necessary packages:
+
+```bash
+sudo apt install rpcbind nfs-kernel-server
+```
+
+Edit `/etc/exports` and add the shares:
+
+```
+/home @myclients(rw,sync,no_subtree_check)
+/usr/local @myclients(rw,sync,no_subtree_check)
+```
+
+The example above shares `/home` and `/usr/local` to all clients in the `myclients` netgroup.
+
+```
+/home 192.168.0.10(rw,sync,no_subtree_check) 192.168.0.11(rw,sync,no_subtree_check)
+/usr/local 192.168.0.10(rw,sync,no_subtree_check) 192.168.0.11(rw,sync,no_subtree_check)
+```
+
+The example above shares `/home` and `/usr/local` to two clients with static IP addresses. If you want instead to allow access to all clients in the private network falling within a designated IP address range, consider the following:
+
+```
+/home 192.168.0.0/255.255.255.0(rw,sync,no_subtree_check)
+/usr/local 192.168.0.0/255.255.255.0(rw,sync,no_subtree_check)
+```
+
+Here, `rw` makes the share read/write, and `sync` requires the server to only reply to requests once any changes have been flushed to disk. This is the safest option; `async` is faster, but dangerous. It is strongly recommended that you read `man exports` if you are considering other options.
+
+After setting up `/etc/exports`, export the shares:
+
+```bash
+sudo exportfs -ra
+```
+
+You'll want to run this command whenever `/etc/exports` is modified.
+
+### Restart services
+
+By default, `rpcbind` only binds to the loopback interface. To enable access to `rpcbind` from remote machines, you need to change `/etc/conf.d/rpcbind` to get rid of either `-l` or `-i 127.0.0.1`.
+
+If any changes are made, rpcbind and NFS will need to be restarted:
+
+```bash
+sudo systemctl restart rpcbind
+sudo systemctl restart nfs-kernel-server
+```
+
+### Security items to consider
+
+Aside from the UID issues discussed above, it should be noted that an attacker could potentially masquerade as a machine that is allowed to map the share, which allows them to create arbitrary UIDs to access your files. One potential solution to this is IPSec; see also the NFS and IPSec section below. You can set up all your domain members to talk to each other only over IPSec, which will effectively authenticate that your client is who it says it is.
```

Comment:
"see also the NFS and IPSec section below": the section this refers to does not seem to exist.

Reply:
I would just remove that sentence

Reply:
Someone made and I think then deleted the sensible-seeming suggestion to just delete this phrase. I shall do so.